x/ibc/testing/types.go

```diff
@@ -1,16 +1,59 @@
 package testing
 
+import (
+	"strconv"
+)
+
+var (
+	ChannelIDPrefix = "channelid"
+)
+
 // TestConnections is a testing helper struct to keep track of the connectionID, source clientID,
 // and counterparty clientID used in creating and interacting with a connection.
 type TestConnection struct {
 	ID                   string
 	ClientID             string
 	CounterpartyClientID string
+	Channels             []TestChannel
+}
+
+// AddTestChannel appends a new TestChannel which contains references to the port and channel ID
+// used for channel creation and interaction. The channel id and port id format:
+// channelid<index>
+// portid<index>
+func (conn *TestConnection) AddTestChannel() TestChannel {
+	channel := conn.NextTestChannel()
+	conn.Channels = append(conn.Channels, channel)
+	return channel
+}
+
+// NextTestChannel returns the next test channel to be created on this connection
+func (conn *TestConnection) NextTestChannel() TestChannel {
+	portID := "transfer"
+	channelID := conn.ID + strconv.Itoa(len(conn.Channels))
+	return TestChannel{
+		PortID:               portID,
+		ID:                   channelID,
+		ClientID:             conn.ClientID,
+		CounterpartyClientID: conn.CounterpartyClientID,
+	}
+}
+
+// FirstOrNextTestChannel returns the first test channel if it exists, otherwise it
+// returns the next test channel to be created.
+func (conn *TestConnection) FirstOrNextTestChannel() TestChannel {
```

Comment:
```suggestion
func (conn *TestConnection) GetTestChannel() TestChannel {
```

Other options, `GetOrCreateTestChannel`, `FetchTestChannel`

Reply:
I'll do `GetOrFetchTestChannel`, a test channel isn't created. In fact just using this func or `NextTestChannel` would result in referencing a channel that doesn't exist on the chain. I'll update the docs to be more clear with the intention of these funcs